e2etest/l2tests/l2.go

```diff
@@ -434,7 +435,7 @@ var _ = ginkgo.Describe("L2", func() {
 
 				err = metrics.ValidateGaugeValue(1, "metallb_speaker_announced", map[string]string{"node": advSpeaker.Spec.NodeName, "protocol": "layer2", "service": fmt.Sprintf("%s/%s", f.Namespace.Name, svc.Name)}, speakerMetrics)
 				if err != nil {
-					return err
+					return pkgerrors.Wrapf(err, "failed to validate metric metallb_speaker_announced for labels: node: %s, protocol: %s, service: %s speakerMetrics: %s", advSpeaker.Spec.NodeName, "layer2", fmt.Sprintf("%s/%s", f.Namespace.Name, svc.Name), speakerMetrics)
```

Comment:
I believe you can hardcode the protocol. Also, I believe the metrics are already printed as result of kubectl (but it won't hurt having them here). Did you check how the output renders?
I am afraid ginkgo will cut the error at some point.

Reply:
the kubectl metrics are printed and i see the specific metric there, but the parsed map is not printed and i suspect the parse function, that's why i want to print it. i tried testing the parse function for the specific output i get from kubectl with unit tests but didn't find anything. 
i printed this whole thing, it's not too long so i don't think ginkgo will cut it.

Reply:
sorry, you were right, ginkgo cuts the error.